tests/cxx/no_fwd_decls.h

```diff
@@ -0,0 +1,22 @@
+//===--- no_fwd_decls.h - test input file for iwyu -----------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// IWYU should not replace the include with a local forward declaration
+#include "tests/cxx/no_fwd_decls-fwd2.h"
+
+// ClassInHeader is in no_fwd_decls-fwd2.h
+ClassInHeader *fwd;
```

Comment:
`*` binds to the left in IWYU style

Reply:
I don't understand how this test case is different from the `ClassInSrc` one? Include a header containing a forward-decl, use the forward-decl in name only, and then provide a definition of the decl?

There are some subtle rules around analysis of associated headers (in this case `no_fwd_decls.{h,cc}` are associated), they "share" more includes than what is typical, so the results could actually be skewed by this (since the `no_fwd_decls-fwd2.h` header is included both here and in the `.cc` file.

Or did you intend for this to be an associated header for some reason I'm missing?